packages/@tinacms/toolkit/src/plugins/branch-switcher/BranchSwitcher.tsx

```diff
@@ -225,20 +264,41 @@ const BranchSelector = ({
         <div className="min-w-[192px] max-h-[24rem] overflow-y-auto flex flex-col w-full h-full rounded-lg shadow-inner bg-white border border-gray-200">
           {filteredBranchList.map((branch) => {
             const isCurrentBranch = branch.name === currentBranch
+            // @ts-ignore
+            const indexingStatus = branch?.indexStatus?.status
             return (
               <div
-                className={`cursor-pointer relative text-base py-1.5 px-3 flex items-center gap-1.5 border-l-0 border-t-0 border-r-0 border-b border-gray-50 w-full outline-none transition-all ease-out duration-150 hover:text-blue-500 focus:text-blue-500 focus:bg-gray-50 hover:bg-gray-50 ${
-                  isCurrentBranch
-                    ? 'bg-blue-50 text-blue-800 pointer-events-none'
-                    : ''
+                className={`relative text-base py-1.5 px-3 flex items-center gap-1.5 border-l-0 border-t-0 border-r-0 border-b border-gray-50 w-full outline-none transition-all ease-out duration-150 ${
+                  indexingStatus !== 'complete'
+                    ? 'bg-gray-50 text-gray-400 pointer-events-none'
+                    : isCurrentBranch
+                    ? 'cursor-pointer bg-blue-50 text-blue-800 pointer-events-none hover:text-blue-500 focus:text-blue-500 focus:bg-gray-50 hover:bg-gray-50'
+                    : 'cursor-pointer hover:text-blue-500 focus:text-blue-500 focus:bg-gray-50 hover:bg-gray-50'
                 }`}
-                key={branch}
-                onClick={() => onChange(branch.name)}
+                key={branch.name}
+                onClick={() => {
+                  if (indexingStatus === 'complete') {
+                    onChange(branch.name)
+                  }
+                }}
               >
                 {isCurrentBranch && (
                   <BiGitBranch className="w-5 h-auto text-blue-500/70" />
                 )}
                 {branch.name}
+                {(indexingStatus === 'inprogress' ||
+                  indexingStatus === 'unknown') && (
```

Comment:
I would treat unknown as a separate state - its not necessarily an error state but the user probably should know if it is in that state for longer than a few seconds, because that means something is wrong with that branch